mockgcp/common/httptogrpc/service.go

```diff
@@ -23,11 +23,68 @@ import (
 	"google.golang.org/genproto/googleapis/api/annotations"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
 	"k8s.io/klog/v2"
 )
 
 // AddService adds a gRPC service (with all the methods) to the mux.
-func (m *grpcMux) AddService(client any) {
+func (m *grpcMux) AddService(client any, options ...ServiceOption) {
+	var opt serviceOptions
+	for _, option := range options {
+		option(&opt)
+	}
+	m.addServiceWithOptions(client, opt)
+}
+
+type ServiceOption func(*serviceOptions)
+
+func WithServiceName(name string) ServiceOption {
+	return func(o *serviceOptions) {
+		o.ServiceName = name
+	}
+}
+
+type serviceOptions struct {
+	ServiceName string
+}
+
+// addServiceWithOptions adds a gRPC service (with all the methods) to the mux.
+func (m *grpcMux) addServiceWithOptions(client any, options serviceOptions) {
+	if options.ServiceName == "" {
+		options.ServiceName = findServiceName(client)
+	}
+
+	serviceName := protoreflect.FullName(options.ServiceName)
+
+	var matchingServices []protoreflect.ServiceDescriptor
+	protoregistry.GlobalFiles.RangeFiles(func(fd protoreflect.FileDescriptor) bool {
+		services := fd.Services()
+		for i := range services.Len() {
+			service := services.Get(i)
+			if service.FullName() != serviceName {
+				continue
+			}
+			matchingServices = append(matchingServices, service)
+		}
+		return true
+	})
+
+	if len(matchingServices) == 0 {
+		klog.Fatalf("could not find proto service %q", serviceName)
```

Comment:
`klog.Fatalf` inside `findServiceName` (which is called by `AddService`) might be too aggressive if this is used as a library. It would panic the entire application if a service cannot be matched. 

While acceptable for a test binary, consider if returning an error up the stack would be better, although `AddService` currently doesn't return an error.

Reply:
Yeah, I don't love it, but because it's part of mock initialization it feels like if something goes wrong here it is a code bug - there should be no external dependencies (e.g. network calls)